index.js

```diff
@@ -276,23 +276,50 @@ async function run() {
     const desiredCount = parseInt((core.getInput('desired-count', {required: false})));
 
 
-    // Register the task definition
-    core.debug('Registering the task definition');
     const taskDefPath = path.isAbsolute(taskDefinitionFile) ?
-      taskDefinitionFile :
-      path.join(process.env.GITHUB_WORKSPACE, taskDefinitionFile);
-    const fileContents = fs.readFileSync(taskDefPath, 'utf8');
-    const taskDefContents = maintainValidObjects(removeIgnoredAttributes(cleanNullKeys(yaml.parse(fileContents))));
-    let registerResponse;
-    try {
-      registerResponse = await ecs.registerTaskDefinition(taskDefContents).promise();
-    } catch (error) {
-      core.setFailed("Failed to register task definition in ECS: " + error.message);
-      core.debug("Task definition contents:");
-      core.debug(JSON.stringify(taskDefContents, undefined, 4));
-      throw(error);
+        taskDefinitionFile :
+        path.join(process.env.GITHUB_WORKSPACE, taskDefinitionFile);
+
+    const isExistingTaskDef = fs.existsSync(taskDefPath);
+
+    let taskDefArn;
+
+    if (!isExistingTaskDef) {
+      core.debug(`Searching for task definition ${taskDefinitionFile} in ECS`);
+      try {
+        const describeResponse = await ecs.describeTaskDefinition({
+          taskDefinition: taskDefinitionFile
+        }).promise();
```

Comment:
This PR needs to be updated to use AWS SDK for JavaScript (v3)

Refer: https://github.com/aws-actions/amazon-ecs-deploy-task-definition/pull/529

Reply:
@trivikr ok